store/sqlstore/utils_test.go

```diff
@@ -76,3 +76,61 @@ func TestSanitizeSearchTerm(t *testing.T) {
 	result = sanitizeSearchTerm(term, "*")
 	require.Equal(t, result, expected)
 }
+
+func TestRemoveNonAlphaNumericUnquotedTerms(t *testing.T) {
+	const (
+		sep           = " "
+		chineseHello  = "你好"
+		japaneseHello = "こんにちは"
+	)
+
+	terms := ""
```

Comment:
Given that we have multiple test cases in the same Test I'll go with structured testing cases like [this example](https://github.com/mattermost/mattermost-server/blob/ad68af10dfcf9d973349f89d025cb5729771a0bf/model/utils_test.go#L51-L72) so we can improve readability and addition of new tests if needed

Reply:
Thanks @ethervoid. 
Table driven test is actually better option here.
I have done the [changes](https://github.com/mattermost/mattermost-server/pull/14884/commits/88ca4510286fe296557d1810d9cf67f39065636b).